fsspec/implementations/reference.py

```diff
@@ -1100,8 +1100,28 @@ def _dircache_from_items(self):
             self.dircache[par].append({"name": path, "type": "file", "size": size})
 
     def _open(self, path, mode="rb", block_size=None, cache_options=None, **kwargs):
-        data = self.cat_file(path)  # load whole chunk into memory
-        return io.BytesIO(data)
+        part_or_url, start0, end0 = self._cat_common(path)
+        if isinstance(part_or_url, bytes):
+            return io.BytesIO(part_or_url[start0:end0])
+
+        protocol, _ = split_protocol(part_or_url)
+        if start0 is None and end0 is None:
+            return self.fss[protocol]._open(
```

Comment:
Ideally, we could put these in `ReferenceFile`, but I did not want to go through an indirection that's not required.

Reply:
This is fine. Maybe add a comment to say what it's doing.

Reply:
Added a small. PTAL.